SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialReceiver.cpp

```diff
@@ -298,12 +301,24 @@ void USpatialReceiver::HandleActorAuthority(Worker_AuthorityChangeOp& Op)
 	{
 		// Check to see if we became authoritative over the UnrealClientRPCEndpoint component over this entity
 		// If we did, our local role should be ROLE_AutonomousProxy. Otherwise ROLE_SimulatedProxy
-		const FClassInfo& Info = ClassInfoManager->GetOrCreateClassInfoByClass(Actor->GetClass());
-
 		if ((Actor->IsA<APawn>() || Actor->IsA<APlayerController>()) && Op.component_id == SpatialConstants::CLIENT_RPC_ENDPOINT_COMPONENT_ID)
 		{
 			Actor->Role = (Op.authority == WORKER_AUTHORITY_AUTHORITATIVE) ? ROLE_AutonomousProxy : ROLE_SimulatedProxy;
 		}
+
+		if (Op.authority == WORKER_AUTHORITY_AUTHORITATIVE)
+		{
+			SpatialGDK::RPCsOnEntityCreation* QueuedRPCs = StaticComponentView->GetComponentData<SpatialGDK::RPCsOnEntityCreation>(Op.entity_id);
+			if (QueuedRPCs != nullptr && QueuedRPCs->HasRPCPayloadData())
+			{
+				ProcessQueuedActorRPCsOnEntityCreation(Actor, QueuedRPCs);
+			}
+
+			ClientRPCEndpoint Endpoint = {};
+			Endpoint.ClientProcessedRPCsOnEntityCreation = true;
+			Worker_ComponentUpdate Update = Endpoint.CreateClientRPCEndpointUpdate();
```

Comment:
We generally have a helper method on our schema classes called `CreateXUpdate`. So you can just do
Worker_ComponentUpdate Update = ClientRPCEndpoint::CreateClientRPCEndpointUpdate(pass in stuff here). Can you do this here as well?

Reply:
Removed this code.